src/public/meeting.ts

```diff
@@ -111,6 +111,30 @@ export namespace meeting {
     isAppSharing: boolean;
   }
 
+  export interface IParticipantSpeakingState {
+    /**
+     * id of participant
+     */
+    id: string;
+
+    /**
+     * indicates whether a participant is speaking
+     */
+    isSpeaking: boolean;
+  }
+
+  export interface ISpeakingState {
+    /**
+     * indicates whether one or more participants in a meeting are speaking
+     */
+    isSpeakingDetected: boolean;
+
+    /**
+     * returns speaking state of participants
+     */
+    participantSpeakingStates: [IParticipantSpeakingState] | undefined;
```

Comment:
`[IParticipantSpeakingState]` means an array with a single element of type `IParticipantSpeakingState`. What I think you wanted to do is IParticipantSpeakingState[]`.

Also, do we need undefined? Why not just empty array for cases where there are no states?

Reply:
Oh yes, thank you for catching that! And empty array works well too -- will update it.

Reply:
Updated and also made this optional, as we would only be returning it if the developer specified to include participants and they have the necessary permission.

Reply:
Heh, looks like my syntax goof carried over. My first serious programming language was Swift, where the `[T]` is a perfectly valid shorthand declaration 😂

Reply:
@ryanbliss all good! I didn't notice until Marcin pointed it out, my bad too 😄